### PR TITLE
Fixed mirrord-agent/cli protocol negotation

### PR DESCRIPTION
**Pull request by aviramha**

### CODE REVIEW

File: mirrord/agent/src/entrypoint.rs [hunk 1]
```diff
@@ -454,15 +453,19 @@ impl ClientConnectionHandler {
                 ))
                 .await?;
             }
-            ClientMessage::SwitchProtocolVersion(version) => {
+            ClientMessage::SwitchProtocolVersion(client_version) => {
+                let our_version = mirrord_protocol::VERSION.get().clone();
```
Comment (DmitryDodzin) [inline mirrord/agent/src/entrypoint.rs:457]: Need also to have some initial value of `1.2.1` as `protocol_version` variable above

Comment (DmitryDodzin) [reply mirrord/agent/src/entrypoint.rs:457]: Also don't need get it's a LazyLock

**Review by DmitryDodzin — CHANGES_REQUESTED** [review]
Still need `1.2.1` version to start with & fix compilation in agent =]

File: mirrord/agent/src/entrypoint.rs (hunk 1 above)
Comment (aviramha) [reply mirrord/agent/src/entrypoint.rs]: got it - I think we can remove it since the switch protocol exists for 10 months already

Comment (DmitryDodzin) [reply mirrord/agent/src/entrypoint.rs]: 😅

Comment (aviramha) [reply mirrord/agent/src/entrypoint.rs]: I'm just afraid that having a base which is a very "old" version while none actually uses it (according to metrics) will create hard bugs (why is it this version??)

**Review by DmitryDodzin — APPROVED** [review]
Alright, Would be strange when return value of negotiation downgrades version rather than upgrading it from older version, though I can't think of any current problems with it just throwing it out there

**Comment by aviramha** [comment]
@DmitryDodzin I thought about it too, but essentially that's how protocol neg should work 🤣 
also, this is what we do in the operator handling side (where we even emulate the message!)

**Pull request merged**